microsetta_public_api/repo/_beta_repo.py

```diff
@@ -0,0 +1,39 @@
+from microsetta_public_api.repo._base import DiversityRepo
+from microsetta_public_api.exceptions import UnknownID, InvalidParameter
+import numpy as np
+
+
+class BetaRepo(DiversityRepo):
+
+    def exists(self, sample_ids, metric):
+        distance_matrix = self._get_resource(metric)
+        if isinstance(sample_ids, str):
+            return sample_ids in distance_matrix.ids
+        else:
+            existing_ids = set(distance_matrix.ids)
```

Comment:
Would it make sense to cache this `set` with the object to avoid recreating it? THis would then allow the `if` portion of this statement to be O(1) rather than O(n)

Reply:
Actually, it looks like this is already in place by using the `in` [operator](https://github.com/biocore/scikit-bio/blob/master/skbio/stats/distance/_base.py#L780)

Reply:
How about https://github.com/biocore/microsetta-public-api/pull/68/commits/558430afaf593a1f7a2171ec7ee86e4f36a223eb